src/packet-ffxiv.c

```diff
@@ -145,8 +150,15 @@ static int dissect_frame(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, vo
     next_tvb = tvb_new_subset_remaining(tvb, FRAME_HEADER_LEN);
   }
 
-  tcp_dissect_pdus(next_tvb, pinfo, frame_tree, TRUE, BLOCK_HEADER_LEN,
-    get_message_length, dissect_message, data);
+  offset = 0;
+  for(int i = 0; i < header.blocks; ++i) {
```

Comment:
Not `i++`? I would maybe use a `while` loop here though since the actual variable isn't important and it will need to run at least once, and saves from having to guard for `header.blocks` being 0, although I doubt that happens.